### ISSUE TITLE
Should the number smoothed_errors be equal to total observations? (i.e. an anomaly score for each observation)

### ISSUE THREAD

**Issue by gorold** [khundman/telemanom#49]
https://github.com/khundman/telemanom/blob/26831a05d47857e194a7725fd982d5dea5402dd4/telemanom/errors.py#L61-L64

I'm wondering if the length of `self.e_s` should be equal to the total number of observations in the sequence? All `self.e_s` are missing 260 observations. 

The above code snippet does not add the 250 observations to the front of the array, but simply changes the existing 250 front elements. Something like this will add them: `self.e_s = np.insert(self.e_s, 0, [np.mean(self.e_s[:self.config.l_s * 2])] * self.config.l_s)` 

However, there is still 10 missing observations from each `self.e_s`, perhaps due to the `n_predictions` option?

**Comment by khundman** [comment]
Thanks for the question, @gorold. `l_s` is the length of the sequence passed to the LSTM, which will use that sequence to start generating predictions for timesteps > `l_s`. Some length `l_s` of input sequence is required to generate future predictions, and because of this the length of the resulting (smoothed) error sequence (`self.e`, `self.e_s`) will be equal to `len(observations) - l_s`. 

The portion of `errors.py` that you referenced isn't intended to restore the sequence to the original observation length. The conversion of early errors to a mean value is to account for large error spikes resulting from the initial predictions that are not smoothed out by the exponentially-weight moving average. If you remove the lines you reference above and re-run you can see this effect.

The additional delta of 10 from the original length of the observations is due to `n_predictions`. We need 10 future `test` values to generate the loss that is backpropagated during training (see https://github.com/khundman/telemanom/issues/24#issuecomment-513411253) and evaluate normalized loss during test. This isn't a requirement for inference however, and I'd welcome a PR to address this.
